sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -247,20 +258,76 @@ func (m *networkFlowManager) Capabilities() []centralsensor.SensorCapability {
 	return nil
 }
 
-func (m *networkFlowManager) Notify(common.SensorComponentEvent) {}
+func (m *networkFlowManager) Notify(e common.SensorComponentEvent) {
+	switch e {
+	case common.SensorComponentEventCentralReachable:
+		m.resetContext()
+		m.resetLastSentState()
+		m.centralReady.Signal()
+	case common.SensorComponentEventOfflineMode:
+		m.centralReady.Reset()
+	}
+}
 
 func (m *networkFlowManager) ResponsesC() <-chan *message.ExpiringMessage {
 	return m.sensorUpdates
 }
 
-func (m *networkFlowManager) enrichConnections() {
-	ticker := time.NewTicker(time.Second * 30)
+func (m *networkFlowManager) resetContext() {
+	m.ctxMutex.Lock()
+	defer m.ctxMutex.Unlock()
+	if m.cancelCtx != nil {
+		m.cancelCtx()
+	}
+	m.pipelineCtx, m.cancelCtx = context.WithCancel(context.Background())
+}
+
+func (m *networkFlowManager) sendToCentral(msg *central.MsgFromSensor) bool {
+	var msgToSend *message.ExpiringMessage
```

Comment:
The message has to be created earlier in the process (as soon as the processing of the event starts). Otherwise we risk getting the connection back up in the middle of the function call and send an inconsistent Network Flow.